src/js_parser.zig

```diff
@@ -3507,7 +3508,7 @@ pub const Parser = struct {
         //    var __dirname = "foo/bar"
         //    var __filename = "foo/bar/baz.js"
         //
-        if (p.options.bundle or !p.options.features.commonjs_at_runtime) {
+        if (p.options.target == Target.browser or !p.options.transform_only) {
```

Comment:
why is target being passed for this? was the original version of this incorrect? i think the main issue is lowering `__dirname` to a string when bundling.

Reply:
Maybe I should look more closely to when commonjs_at_runtime is true. To exactly match current tooling it seemed best to do this only when target was browser. Also I think in node later I use .dirname while in bun it was set to use "dir". 

So I think the cases are 

* Bun run format!="cjs" ... - always inline
* Bun build --target="node" -> import.meta.dirname / path.basename(pathToFileURL(import.meta.url))
* Bun build --target="bun" -> import.meta.dir
* Bun build --target="browser" -> souce_path
* Bun build format=cjs -> do nothing